WordPress/Classes/ViewRelated/Reader/ReaderFollowedSitesViewController.swift

```diff
@@ -0,0 +1,332 @@
+                let title = NSLocalizedString("Could not Unfollow Site", comment: "Title of a prompt.")
+                let description = error.localizedDescription
+                let buttonTitle = NSLocalizedString("OK", comment: "Button title. Acknowledges a prompt.")
+                let alert = UIAlertController(title: title, message: description, preferredStyle: .Alert)
+                alert.addCancelActionWithTitle(buttonTitle)
+                alert.presentFromRootViewController()
+        }
+    }
+
+
+    func followSite(site: String) {
+        guard let url = urlFromString(site) else {
+            let title = NSLocalizedString("Please enter a valid URL", comment: "Title of a prompt.")
+            let buttonTitle = NSLocalizedString("OK", comment: "Button title. Acknowledges a prompt.")
+            let alert = UIAlertController(title: title, message: "", preferredStyle: .Alert)
+            alert.addCancelActionWithTitle(buttonTitle)
+            alert.presentFromRootViewController()
+            return
+        }
+
+        let service = ReaderSiteService(managedObjectContext: managedObjectContext())
+        service.followSiteByURL(url, success: { [weak self] in
+            let success = NSLocalizedString("Followed", comment: "User followed a site.")
+            SVProgressHUD.showSuccessWithStatus(success)
+            self?.syncSites()
+
+        }) { (error) in
+            DDLogSwift.logError("Could not follow site: \(error)")
+            let title = NSLocalizedString("Could not Follow Site", comment: "Title of a prompt.")
+            let description = error.localizedDescription
+            let buttonTitle = NSLocalizedString("OK", comment: "Button title. Acknowledges a prompt.")
+            let alert = UIAlertController(title: title, message: description, preferredStyle: .Alert)
+            alert.addCancelActionWithTitle(buttonTitle)
+            alert.presentFromRootViewController()
+        }
+    }
+
+
+    func urlFromString(str: String) -> NSURL? {
+        // if the string contains space its not a URL
+        if str.containsString(" ") {
+            return nil
+        }
+
+        // if the string does not have either a dot or protocol its not a URL
+        if !str.containsString(".") && !str.containsString("://")  {
+            return nil
+        }
+
+        var urlStr = str
+        if !urlStr.containsString("://") {
+            urlStr = "http://\(str)"
+        }
+
+        if let url = NSURL(string: urlStr) where url.host != nil {
+            return url
+        }
+
+        return nil
+    }
+
+
+    func showPostListForSite(site: ReaderSiteTopic) {
+        let controller = ReaderStreamViewController.controllerWithTopic(site)
+        navigationController?.pushViewController(controller, animated: true)
+    }
+}
+
+
+extension ReaderFollowedSitesViewController : WPTableViewHandlerDelegate
+{
+
+    func managedObjectContext() -> NSManagedObjectContext! {
```

Comment:
This might be nicer as a computed property instead of a method, and return a non-optional context:

```
var managedObjectContext: NSManagedObjectContext {
    return ContextManager.sharedInstance().mainContext
}
```